modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ApplicationModel.java

```diff
@@ -337,24 +337,22 @@ public ApplicationModel(ArtifactConfig artifactConfig, ArtifactDeclaration artif
     topLevelComponentsStream()
         .forEach(componentModel -> ((ComponentModel) componentModel).resolveTypedComponentIdentifier(extensionModelHelper,
                                                                                                      runtimeMode));
-    final ComponentLocationVisitor clv = new ComponentLocationVisitor();
-    recursiveStreamWithHierarchy().forEach(clv);
+    recursiveStreamWithHierarchy().forEach(new ComponentLocationVisitor());
   }
 
   public void macroExpandXmlSdkComponents(Set<ExtensionModel> extensionModels) {
     expandModules(extensionModels, () -> {
       // TODO MULE-13894 do this only on runtimeMode=true once unified extensionModel names to use camelCase (see smart connectors
       // connectors and crafted declared extension models)
+
+      // TODO MULE-17419 (AST) Remove these 2 actions
```

Comment:
If we don't resolveComponentTypes those componentAst will not be associated to a parameter model etc therefore it will not be able to call getParameterAst by the parsers.

Reply:
if those are already resolved for the AST of the smart connector definition, the idea is to keep and use them. In that case, there's no need to recalculate them here. That's one of the golas of the linked TODO.

Reply:
👍